worker/uniter/runner/jujuc/state-cache-set.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package jujuc
+
+import (
+	"bufio"
+	"io"
+	"strings"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"github.com/juju/utils/keyvalues"
+
+	jujucmd "github.com/juju/juju/cmd"
+)
+
+// StateCacheSetCommand implements the state-value-set command.
+type StateCacheSetCommand struct {
+	cmd.CommandBase
+	ctx          Context
+	StateValues  map[string]string
+	keyValueFile cmd.FileVar
+}
+
+func NewStateCacheSetCommand(ctx Context) (cmd.Command, error) {
+	return &StateCacheSetCommand{ctx: ctx}, nil
+}
+
+func (c *StateCacheSetCommand) Info() *cmd.Info {
+	doc := `
+state-cache-set sets the value of the server side state specified by key.
+Setting an empty string causes the setting to be removed.
+
+The --file option should be used when one or more key-value pairs are
+too long to fit within the command length limit of the shell or
+operating system. The file will contain a YAML map containing the
+settings.  Settings in the file will be overridden by any duplicate
+key-value arguments. A value of "-" for the filename means <stdin>.
+
+See also:
+    state-cache-delete
+    state-cache-get
+`
+	return jujucmd.Info(&cmd.Info{
+		Name:    "state-cache-set",
+		Args:    "key=value [key=value ...]",
+		Purpose: "set server-side-state values",
+		Doc:     doc,
+	})
+}
+
+func (c *StateCacheSetCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.keyValueFile.SetStdin()
+	f.Var(&c.keyValueFile, "file", "file containing key-value pairs")
+}
+
+func (c *StateCacheSetCommand) Init(args []string) error {
+	if args == nil {
+		return nil
+	}
+
+	// The overrides will be applied during Run when c.keyValueFile is handled.
+	overrides, err := keyvalues.Parse(args, true)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	c.StateValues = overrides
+	return nil
+}
+
+func (c *StateCacheSetCommand) Run(ctx *cmd.Context) error {
+	if err := c.handleKeyValueFile(ctx); err != nil {
+		return errors.Trace(err)
+	}
+
+	for k, v := range c.StateValues {
+		if v != "" {
+			if err := c.ctx.SetCacheValue(k, v); err != nil {
+				return err
+			}
+			continue
+		}
+
+		if err := c.ctx.DeleteCacheValue(k); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *StateCacheSetCommand) handleKeyValueFile(ctx *cmd.Context) error {
+	if c.keyValueFile.Path == "" {
+		return nil
+	}
+
+	file, err := c.keyValueFile.Open(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer func() { _ = file.Close() }()
+
+	kvs, err := readSettings(file)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	overrides := c.StateValues
+	for k, v := range overrides {
+		kvs[k] = v
+	}
+	c.StateValues = kvs
+	return nil
+}
+
+func (c *StateCacheSetCommand) readKeyValuePairs(in io.Reader) (map[string]string, error) {
```

Comment:
This method is redundant since the above code uses `readSettings` now.

Reply:
removed.